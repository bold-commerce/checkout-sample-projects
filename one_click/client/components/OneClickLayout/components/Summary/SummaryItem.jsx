import React from "react";
import TimesCircle from "../Icons/TimesCirlce";
import { Price } from "@boldcommerce/stacks-ui/lib";

const SummaryItem = ({ title, value, removeItem }) => {
    return (    
        <div className="summary-line-child" >
            <div className="summary-child-title">
                { title }
                { removeItem &&
                <button
                    className="summary-child-remove"
                    onClick={() => removeItem(title)}
                >
                    <TimesCircle />
                </button>
                }
            </div>
            <Price className="summary-child-value" amount={value} />
        </div>
    )
}

export default SummaryItem;