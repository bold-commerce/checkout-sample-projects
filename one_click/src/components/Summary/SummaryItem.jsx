import React from "react";
import TimesCircle from "../Icons/TimesCircle";
import { Price } from "@boldcommerce/stacks-ui/lib";
import { useTranslation } from 'react-i18next';

const SummaryItem = ({ title, value, removeItem }) => {
    const { t } = useTranslation();
    return (    
        <div className="summary-line-child" >
            <div className="summary-child-title">
                { title }
                { removeItem &&
                <button
                    aria-label="remove item"
                    className="summary-child-remove"
                    onClick={() => removeItem(title)}
                >
                    <TimesCircle />
                </button>
                }
            </div>
            <Price className="summary-child-value" amount={value} moneyFormatString={t('currency_format')} />
        </div>
    )
}

export default SummaryItem;