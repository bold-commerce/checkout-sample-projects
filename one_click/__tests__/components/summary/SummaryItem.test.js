import React from 'react';
import { render } from '@testing-library/react';
import SummaryItem from '../../../client/components/OneClickLayout/components/Summary/SummaryItem';

describe('SummaryItem', () => {
    test('renders SummaryItem component', () => {
        const { asFragment } = render (
            <SummaryItem
                title='test title'
                value={1991}
            />
        )

        expect(asFragment()).toMatchSnapshot();
    });

    test('renders SummaryItem component that is removable', () => {
        const { asFragment } = render(
            <SummaryItem
                title='test title two'
                value={2121}
                removeItem={() => {}}
            />
        )
        expect(asFragment()).toMatchSnapshot();
    });
});