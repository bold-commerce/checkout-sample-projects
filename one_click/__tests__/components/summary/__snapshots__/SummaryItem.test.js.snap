// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryItem renders SummaryItem component 1`] = `
<DocumentFragment>
  <div
    class="summary-line-child"
  >
    <div
      class="summary-child-title"
    >
      test title
    </div>
    <div
      class="stx-price summary-child-value stx-price--alight-center"
    >
      <p
        aria-label="Product price"
        class="stx-price__amount"
      >
        CA$19.91
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SummaryItem renders SummaryItem component that is removable 1`] = `
<DocumentFragment>
  <div
    class="summary-line-child"
  >
    <div
      class="summary-child-title"
    >
      test title two
      <button
        aria-label="remove item"
        class="summary-child-remove"
      >
        <svg
          aria-labelledby="button"
          fill="none"
          height="20"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            opacity="0.8"
          >
            <path
              clip-rule="evenodd"
              d="M17.7071 7.70706L7.70712 17.7071L6.29291 16.2928L16.2929 6.29285L17.7071 7.70706Z"
              fill="black"
              fill-rule="evenodd"
            />
            <path
              clip-rule="evenodd"
              d="M16.2929 17.7071L6.29291 7.70706L7.70712 6.29285L17.7071 16.2928L16.2929 17.7071Z"
              fill="black"
              fill-rule="evenodd"
            />
            <path
              clip-rule="evenodd"
              d="M12 3C7.02944 3 3 7.02944 3 12C3 16.9706 7.02944 21 12 21C16.9706 21 21 16.9706 21 12C21 7.02944 16.9706 3 12 3ZM1 12C1 5.92487 5.92487 1 12 1C18.0751 1 23 5.92487 23 12C23 18.0751 18.0751 23 12 23C5.92487 23 1 18.0751 1 12Z"
              fill="black"
              fill-rule="evenodd"
            />
          </g>
        </svg>
      </button>
    </div>
    <div
      class="stx-price summary-child-value stx-price--alight-center"
    >
      <p
        aria-label="Product price"
        class="stx-price__amount"
      >
        CA$21.21
      </p>
    </div>
  </div>
</DocumentFragment>
`;
